src/components/DeviceInteraction/DeviceInteractionStep.js

```diff
@@ -120,8 +124,21 @@ class DeviceInteractionStep extends PureComponent<
     try {
       const d1 = Date.now()
 
-      // $FlowFixMe JUST TESTED THE `run` 6 LINES BEFORE!!!
-      const res = (await step.run(data)) || {}
+      let res
+
+      if (step.timeout) {
+        const timeoutPromise = new Promise(r => setTimeout(() => r('timeout'), step.timeout))
```

Comment:
- you can do `r(new TimeoutError())` directly
- the timeout is never cancelled which i think might be a problem. there is two case we could cancel it: if step unmount, if job finishes. 

> BTW it would be much better if later we move the "job" to use observable paradigm, it have all of these out of the box, `.pipe(timeout(..))`